### ISSUE TITLE
first lookup after activation is not producing inflection matches

### ISSUE THREAD

**Issue by balmas** [alpheios-project/inflection-tables#161]
this seems to have been introduced with #160 . 

The first lookup of any query results in the inflection tables after activation of the extension does not identify matches. Initialization of matching seems to be delayed until after the first display. From then on it works as expected. This also seems to occur separately for each language.

**Comment by balmas** [comment]
@kirlat fixed this in 9ca370a11ac0e992e9f82946e189ae723c0388d4 and https://github.com/alpheios-project/components/commit/4413a5b05c3767c386b1a66885a1054e6ee0977a

Explanation copied from Slack:

>This is because suffixes are inserted into a table by reference. With introduction of an inflection browser we have two instances of verb views (if lexical query result is a verb) shown at the same time. Those two views share the same suffix object. Because of this, suffixes share state, including `showMatches`. That's why latest options redefine the earliest one.

First time inflection browser's table is loaded last, so it resets lexical query result's state and both show no matches. Second time, however, inflection browser tables are already generated, and lexical query results table is loaded after inflection browser tables. Its option to show matches override those in inflection browser and we can see that inflection browser starts to show matches too.

The best solution would probably be be to copy suffix items when attaching them to the table so they will be independent. I'm afraid to do that now, however, because suffixes can have complex data attached, and we need to copy every little piece of this data properly. And what data we have can differ from one case to the other. This will be very tricky. And this change will affect all inflection tables, so it will involve a huge amount of retesting. I'm afraid to introduce such a big change that late before the release.

I think for now I can solve it on a representational level (within a Vue component) to force inflection browser show no match classes. We can probably postpone copying for later (refactoring?). What do you think?

response from @balmas:
>Agree to postpone copying for refactoring so for now, the fix is to always have showMatches be true, and to just hide the styles in the browser.

**Comment by balmas** [comment]
@kirlat I don't think this is fixed. At first I thought it was because of this https://github.com/alpheios-project/inflection-tables/commit/9ca370a11ac0e992e9f82946e189ae723c0388d4#r30567141 but I changed that and the matches still are missing the first time the table is displayed.

**Comment by balmas** [comment]
is there any chance this is introduced by something with webpack?  I was pretty sure the first build I ran with it worked, but now it is broken again.

**Comment by balmas** [comment]
hmm. so this bug does seem restricted to verb views. I don't think it's webpack related then, I went back to the build that I thought was working and can still reproduce it. It can be reproduced with verb views only. E.g. activate and lookup 'cepit' and the first time the table comes up the matches isn't there. But if you activate and lookup 'paululum' it's fine.

So it does seem that it must be caused by the shared instances of the Suffix objects in the verb views.

**Comment by balmas** [comment]
but maybe it's not the Suffixes actually that are the problem here. Is it possible that we are getting the wrong version of the view? Maybe something with this code? https://github.com/alpheios-project/inflection-tables/blob/master/views/lib/view-set.js#L60-L72

**Comment by balmas** [comment]
another interesting side-effect of this, when you get the wrong view for a verb, e.g. on the first lookup of 'cepit', and then use the view switcher to see a different variation on the verb table (e.g. conjugation-voice-mood instead of voice-mood-conjugation) all of the cells in the alternate view have the morphologyMatches border. Redo the lookup and then that is fixed.

( I note that with the first lookup, the constructor for the verb table that is displayed is called twice, once for the form in the query results and then again for the standard form. And then when you do the lookup a second time, getting the correct results, it calls the constructor for the query results only. So it would seem to definitely be exposed by the creation of the standard forms version of the table. )

**Comment by monzug** [comment]
installed build 2.0.3-8 and I could see all cells highlighted for latin verb circumlātum  and no exact matches.

**Comment by kirlat** [comment]
Thanks for your information! I'm working on debugging this issue right now.

**Comment by monzug** [comment]
@Kirlat. Good luck! let me know if I can be of any help. I did install the
latest build 2.0.3-8 in Firefox. as first lookup I entered iremus. I got:
1) a script causes Firefox to shutdown or similar message but Firefox did
not shutdown
2) then clicking on Inflect, all cells highlighted
3) only after the second lookup, I got exact match and the right cells
highlighted

**Comment by balmas** [comment]
this fix seems to work for me. Please issue a PR and merge. Thanks! Will enter a separate issue for the Firefox alert.

**Comment by balmas** [comment]
@monzug please use https://github.com/alpheios-project/webextension/tree/qa-2.0.3-9 to test this. Thanks!

**Comment by monzug** [comment]
manifest file is missing - should I install from archive?

**Comment by monzug** [comment]
it did go the second time - all kosher!

**Comment by monzug** [comment]
Chrome, build 2.0.3-9  
Lookup a Greek noun and a greek verb and both showed the exact matches and highlighted cells. will keep testing

**Comment by monzug** [comment]
installed in Firefox, got again the message reported in issue #170
tested with few latin and greek words, so far so good
